scripts/openapi.py

```
@@ -0,0 +1,133 @@
+import urllib.request
+import urllib.error
+from pathlib import Path
+
+
+def wait_for_server(url: str, max_wait: int = 300) -> bool:
+    """Wait for the server to become ready."""
+    elapsed = 0
+    sleep_time = 1
+    while elapsed < max_wait:
+        try:
+            with urllib.request.urlopen(url, timeout=5) as resp:
```

Ruff (0.15.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)